src/GraphQL.Tests/Types/SchemaTests.cs

```diff
@@ -160,6 +162,50 @@ private void DoesNotContainTypeNames(Schema schema, params string[] typeNames)
                 type.ShouldBe(null, $"Found {typeName} in type lookup.");
             });
         }
+
+        [Fact]
+        public void middleware_can_reference_SchemaTypes()
+        {
+            var schema = new Schema { Query = new SomeQuery() };
+            schema.FieldMiddleware.Use(next =>
+            {
+                schema.AllTypes.Count.ShouldNotBe(0);
+                return async context =>
+                {
+                    var res = await next(context);
+                    return "One " + res.ToString();
+                };
+            });
+            schema.Initialize();
+        }
+
+        [Fact]
+        public void disposed_schema_throws_errors()
+        {
+            var schema = new Schema();
+            schema.Dispose();
```

Comment:
```suggestion
            schema.Dispose();
            schema.Dispose();
```

Just to test idempotency.